### DISCUSSION TITLE
Djot and Markdown interoperability ?

### DISCUSSION THREAD

**Discussion by simonmichael** [jgm/djot#219] [General]
Thinking about djot becoming a practical reality for myself and in the tools we all use (Obsidian.. Github..), and how long it might take... I wondered how well djot users could function while using those existing markdown tools, and how well those tools could cope with djot. I tried a djot readme in Obsidian and it looked fine except for also showing all the {#blockids} in the rendered output.  What's your experience ?

New standards that gracefully interoperate with the old tend to do very well with adoption. See eg Typescript. djot is not a strict subset of markdown, and maybe can't be. But is there more of an interoperability story possible ? Eg, 

- Is it worth documenting a subset of djot ("djot lite") that can be used safely in markdown tools until they support full djot ?

- What minimal support or optional modes could markdown tools add to become reasonably usable for djot ? (Such as, ignoring curly-braced text that looks like djot inline/block attributes ?)

**Comment by waldyrious** [comment]
> Is it worth documenting a subset of djot ("djot lite") that can be used safely in markdown tools until they support full djot ?

This seems interesting. In practice it would be a subset of both djot and markdown, i.e. an intersection (one could also think of it as e.g. "sane markdown" :smile:). I'd be curious to see that as well.

> Such as, ignoring curly-braced text that looks like djot inline/block attributes?

Just for the fun of it, I wonder if a backwards-compatible Djot would be possible by using the HTML comment syntax for the metadata/attribute bits, e.g. `<!--{#blockid}-->`/`<!--{key="value"}-->` instead of `{#blockid}`/`{key="value"}`.

**Reply by bdarcus** [reply]
Almost all of the basic markdown I use in, say, GitHub, is also already `djot`.

But I think the point of the new language is to avoid the sort of compromises and inconsistencies of markdown.

From that POV, the HTML comment syntax seems to me a compromise I wouldn't want to accept myself.

**Reply by waldyrious** [reply]
Well, I suppose the mere fact that `*foo*` is interpreted as italics in markdown and as bold in djot is already a deal-breaker for such an attempt anyway. It still was a fun thought to entertain :)

**Reply by bdarcus** [reply]
Yeah, I forgot about that one!

**Comment by Conaclos** [comment]
I wrote a [minimal guide](https://github.com/lib9/guides/blob/main/lib9-markdown-style-guide.md#use-basic-markdown) for writing _djot_ and _commonmark_ compatible code.

The only incompatibility is the table heading line: Markdown formatters, such as prettier, add spaces between `-` and `|`.
See #217 for more details.

**Comment by dereuromark** [comment]
I wouldnt worry about this. Too impractical.
Instead editors (via plugins) should always have import/export capabilities.

**Comment by jgm** [comment]
pandoc can convert between djot and various dialects of markdown.

**Reply by dereuromark** [reply]
Yes, that's sure another possibility besides native editor import/export.
My approach, if built in, usually is a single click though.

However, under the hood, they likely could be using pandoc as tool.
